**MICB 475 2024W2 Team 1**

# Mar 27 2025 Team Meeting

### Agenda:
- Going over slides made:
  https://docs.google.com/presentation/d/1gByiG3n8KTYBurb32oyNeu1aBkXKJJZ9DSyT2nKYX44/edit?usp=sharing

### Meeting Minutes:

#### Aim 1: Diversity analysis
- Shannon and Simpson's are same analysis but different scale - probably why we got the same p-value
- Results: Abundance - increase, Phylogenetic - decrease
- More diverse when comes to evenness, less diverse when comes to phylogenetic distance (spaceflight samples more closely related to each other)
- For Faith's, we can say there's a downward trend (one outlier data point that is affecting significance)
- For Beta analysis: 3 for ground is a big limitation (result of rarefaction) -> Do permanova (wilcoxson's doesn't work for beta)
- Figure 1: A) Observed, B) Simpson or Shannon, C) Faith’s PD
- Figure 2: A) Bray-Curtis, B) Weighted UniFrac

#### Aim 2: Differential abundance analysis
- No significant difference, likely due to small sample size (6 ISS, 6 Earth) reduces statistical power
- So instead generated top 10 most abundant ASVs -> but only resolved to 3 genus levels (Pseudomonas, Stenotrophomonas, NA)
- For the plots, can't use abundance of raw counts -> need to look at average relative abundance
- Figure 3. Taxa bar plots (different taxa levels: genus, family, phylum)

### Aim 3: Functional analysis
- Pathways for breaking down common carbon sources (lactose, galatose, starch, etc.) is downregulated
- But pathways for breaking down niche (or less commonly consumed) carbon sources is upregulated (like toulene)
  - Interesting because these carbon sources are worse (hard to use)
  - Is it because these carbon sources are more readily available in space (or the common ones become less available)?
- AI could help with high-level observations of bar plot showing the top 15 pathways
  - Get rid of the middle 3 pathways that down trend toward upregulation or downregulation
- Figure 4. volcano + top 10 pathways

#### Other notes
- Not worth doing DESEq2 differential abundance analysis anymore
- For presentation & manuscript, center the story around Aim 3 (the most interesting result)
- Make the theme of all the plots match (could assign a colour to each condition)

### Todos:
- [ ] Start on draft manuscript

# Mar 20 2025 Team Meeting

### Agenda:
 - Working on analysis (over halfway done), have questions/want feedback

### Meeting Notes:

#### Aim 1: Diversity analysis
- All good so far
- Set the axis to zero for some of the plots

#### Aim 2: Core microbiome and indicator species analysis
- Prevalence threshold is too low (10%), should scrap this aim
- Can work on Aim 3 (do DESeq2)

#### Aim 3: Differential abundance analysis
- During manuscript writing, mention what we did and suggest alternative methods in future directions
- no significant difference between ISS vs Earth samples so just present the top 10 most abundant ASVs

#### Aim 4: Functional analysis
- Need to filter dataset first, to exclude samples where Spaceflight is Not Applicable
- Good results to work with, could look into sex differences

### Todos:
- [ ] Complete analysis
- [ ] Create presentation

# Mar 14 2025 Team Meeting

### Agenda:
- Submitted revised proposal
- Working on analysis, will ask questions/get feedback (depending on how much has been completed)

### Meeting Minutes:
- if have lots of pairwise comparisons, then do false discovery rates(?) but this doesn't apply to us
- Aldex2 code in progress
- documentation notes -> copy and paste into github
  - separate them by figures (each figure has the code used to generate figure)
- just need good format + structure + level of detail
- differential analysis -> boxplot vs heat map
  - set 1 as standard, 1 as experimental, heat map is good but might be overkill for 2 conditions

### Todos:
- [ ] have all data generated by next week

# Mar 7 2025 Team Meeting

### Agenda:
- Received proposal back so working on revising it this week
- Cover questions related to the proposal
- Dicuss edits per Hans' suggestions

### Meeting Minutes:
- For bioinfo work, should always cite all the tools used
- Microbiotia vs Microbiome: Microbiota is organisms while microbiome is genome but often these two terms are interchanged in literature
- Intro: First sentence with trivia fact is not as closely related to DFUs - can be removed as the next sentence provides enough context
- Hypothesis & Aim 4: Try to find other studies to back up the observations about Pseudomonas and Morganella in the original paper, if there aren't any then mention that there are limited studies in this field
- Grading: Based on whether we put effort into implementing revisions (don't necessarily need to agree with all suggestions)

### Todos:
- [ ] Work on revision edits and check over proposal by Mar 10

# Feb 28 2025 Team Meeting

### Agenda:

#### Proposal completed:
- PDF link: https://drive.google.com/file/d/1d8Pd0KtT512UcPfwP6cnKbmrvWJpILS0/view?usp=sharing
- Get feedback from Hans

#### Task delegation for manuscript:
- Aim 1 --> Sam
- Aim 2 --> Ada
- Aim 3 --> Jen/Byron 
- Aim 4 --> Troy

### Meeting Minutes:

#### Reminders:
- Don't have to stick to proposal
- Can resubmit the proposal after it’s been graded
  - Potential to gain 5 points back as long as we take the time to implement the TA's suggested changes

#### How should the instruction be structured?
- Not a lot of work being done on diabetic foot ulcers - hard to find existing literature
- Looked for articles to back up the obvious (i.e., comparing ground vs space samples)
- **Suggestions**:
  - Start with microbiome, then mention impact of diabetes on the body
  - See if there’s any literature that shows how there are physiological changes due to diabetes and how that affects the microbiome.
    - Ex: Increasing body temperature, skin changes
  - Look at how diabetes normally affects the body and whether disease progression/affect on the body changes in space
  - See if there’s papers explaining how space affects the body in general
    - Ex: Less movement → muscle atrophy
    - Ex: No showers → impact immunity
  - Then consider how does that affect the health status and microbiome
- **Takeaway**: If we can't find specific papers, need to make the links about the impact of our dataset analysis ourselves

#### Should we use ALDEx2?
- ANCOM is the same thing as ALDEx2, could use ANCOM instead
  - more people use ANCOM
  - Works in similar way in that it takes log ratios of microbes
  - Difference between these two methods is based on the control phylum used (maybe, need to check how it works)
- **Takeaway**: If there isn’t enough info on ALDEx2, it’s better to switch to ANCOM to ease the work burden of figuring out ALDEx2 on our own
  - Can switch at any point while working on the project

#### Proposed timeline & workflow (based on previous students’ experiences)
- At this stage there are 4 weeks until presentation week, so 4 meetings left: Mar 7, 14, 21, 28
  - Mar 21: Bring completed figures and ideas for presentation
  - Mar 28 (before presentation): Should have analysis done, bring a plan of the presentation (prep slides)
- Figures/results & story - most of the manuscript
- As we get the data from each analysis step, start making the panels for each aim - create them as they would appear in the manuscript
- Also try to include some bullet points for the results write-up

### Todos:
- [ ] Analysis kinda be done - Mar 14
- [ ] Analysis completely done - Mar 21 
- [ ] Slides done - Mar 28

# Feb 19 2025 Team Meeting

### Agenda:
- Delegate roles for proposal and discuss any concerns

### Meeting Minutes:

#### Rough outline decided:
- Wed Feb 19 2:45pm meeting w/ Hans
  - data wrangling should be done
  - very rough drafts of our own parts should be done
- Sat Feb 22 - final draft ready
- Sun Feb 23 - meeting to finalize proposal + submit

#### Proposal delegation:
- Title --> group
- Introduction and Background --> Jen + Sam
- Research Objective --> Troy + Ada
- Experimental Aims and Rationale
  - aim 1 --> Sam
  - aim 2 --> Jen
  - aim 3 --> Ada
  - aim 4 --> Troy
  - data wrangling --> Byron
- Overview Chart --> group
- Timeline --> Troy
- Participation Report --> group
- References --> group

#### Data wrangling
- data wrangling notes updated in module_notes.m
- Paired end data, but class was single-end (changed the code)
- Line endings were a little different -> moved it into an editable file
- Closely in relation to class activities
- Demux File: average phred schore of 30
- QUESTION: How to trim? Reverse is better quality than forward (usually the opposite)
- 205 -> want an average phred score 30 or up
- Find the median percentile (look at the quality score at that position) -> still pretty high (keep the whole thing?)
  - Cut it like 230
  - Cutting it like 50 bps (forward and reverse do not have to be the same)
  - Cut it to be a bit longer -> (reasoning is alright, but if we cut out 1/5 of our entire sequence we are losing a lot of resolution, so more uncertainity)
  - Server has been clogged and a lot of backlog (took a longtime)
- Files too large to store on GitHub (store the latest ones on there) -> store it 

### Todos:
- [ ] Complete individual parts of proposal

# Feb 13 2025 Team Meeting
### Agenda:

**Research paper**: https://pmc.ncbi.nlm.nih.gov/articles/PMC9761711/

**Dataset**: diabetic ulcers dataset

#### Things to cover:
- Discuss concerns w/ dataset:
- Discuss and pick specific research Q w/ this dataset
- Discuss what we'd like to explore (3-4 aims)

#### Concerns:
  - small sample size, how can we cluster?
  - some of the metadata have a lot of NA fields

#### Research question ideas:
  - Does sex influence the microbial composition of foot ulcers? (exclude the ones in space)
  - Only do the skin swabs
  - health vs ulcer (added element of sex difference)

#### Aim Ideas:
  1. Does sex influence the microbial composition of foot ulcers? (skin swabs only)
  2. Does space conditions affect microbial composition of foot ulcers? (tissue samples only, space vs ground control)
  3. Does sex influence the microbial composition of foot ulcers in space conditions? (tissue samples only, space vs ground control)
  4. Functional analysis (healthy vs DFUs samples)
     - disease progression? or potential?
     - biofilm formation
     - compare with a reference genome and identify differences -> if possible describe general functions/lack of function
       - eg. we noticed that these species were missing from these samples. could affect disease potential/progression?

### Meeting Minutes:
- received data from Dr. Sun

#### Discussing ulcers dataset:
- Concerned about the small sample size
  - how did they cluster with such a small amount?
  - if we subset to only the space flight experiment samples: 6 samples that remained on Earth and 6 samples sent to space
  - All together -> would have few samples
- Struggling to connect all the different samples
- Separating the different types into one product or question (subset of patients)
- Can't cluster by sex -> too small of a sample size
- Even separating from sample sizes -> categories are the same (no difference)

#### Options to address concerns:
- Not much to work with
- find something to make it work -> pivot to another dataset
- OR change the aims:
  - AIM 1 -> replicating the data that was already shown by the authors (similar) -> replicate their diversity metrics (COULD ADD MORE -> FOCUS ON SPACE VS. EARTH)
  - AIM 2 -> (get rid of sex) replicate the abundance analysis but adjust the methods, use a different package for analysis
    - NEW ONES SENT BY HANS: Aldex2 and MICOM
    - These two packages don't take relative abundance of a number of reads -> use ratios between the same microbes across samples -> will remove the bias of the 16s sequences
    - 16s relative abundance is not a good metric, b/c biased to abundance
    - Log of ratios. Take the ratio of 2 phyla and the log of that ratio and calculate it across samples will give better information about the differential abundance. Slightly less biased information of microbes that are truly differentially abundant between the two samples.
    - So we will be replicating the author's analysis (they only looked at relative abundance), but using a different method. Can directly compare to their findings. Use both tools. Spin it into a methodology problem
  - Overall want 6 samples per group is enough considering the tough conditions
    - Not possible to subset by sex because it would be 2 vs 4 -> problem, so want to have at least 3 per subgroup
  - AIM 3 - core microbiome and indicator species
  - AIM 4 - functional analysis and aspect of the same 2 groups (space vs nonspace) -> extrapolate who is there and base the analysis over that -> see the next closest genome (picrust2)

#### More discusion about how to work with ulcers dataset:
- With the new aims, can generate at least 4 figures
- Have four aims that only target the 6 samples sent to sapce + their ground controls
- If we wanted to directly replicate results -> would have multiple pairwise comparisons because they are distinct
- A lot of analysis of matched samples
- Focus on good methodology -> expanding that same code to different samples
- Focus on space vs. earth (confirm the diversity and expand it by choosing different diversity metrics)
- Inspiration: https://huttenhower.sph.harvard.edu/tools/ -> all tools on 16s RNA analysis
- Introduction = Brief description of what is known and broad -> planning what we are doing. Brief overview of knowledge and literature
- Research Objective -> Aims 

### Todos:
- [ ] RUN CODE THROUGH QIIME (filtering step, include only what we are working with) 
- [ ] HAVE STEPS ON HOW TO ACHIEVE ALL OF THE AIMS (STEP BY STEP WORKFLOW) -> do what we need to do and if there are any changes

# Feb 7 2025 Team Meeting

### Agenda:

#### Interested datasets:
- alcohol dependency
- multiple sclerosis
- vaping dataset
- ISS

#### Potential research questions:
- Is it possible to accurately predict an individual’s smoking / vaping status based on their oral microbiome profile? -> Vaping
- Can the gut microbiome be used to predict symptoms of depression?
- In patients with MS, does smoking or former smoking have an impact on microbiota composition and diversity, could this be driving disease progression in MS
- How does microbial diversity + composition on the ISS compare to other confined environments such as simulated space habitats (HI-SEAS)? + college dormitories? -> ISS
- Reasons for choosing smoking-related datasets:
  - WHY? -> smoking is very prevalent in our society
  - Could be a new avenue in preventative medicine 

### Meeting Minutes:

#### Proposed ideas:
- MS -> it's being done by 2 other groups?
- smoking -> predictive model (must be able to find significant differences in the models for it to work)
  - didn't find a diff in vaping vs non-vaping but did find something in smoking
- depression -> another group is doing a predictive model
- microbial diversity on ISS -> already done

#### Some notes:
- predictive models aren't too hard?
- NASA datasets
  - they publish new datasets every day?
  - space radiation dataset not touched
- can only look at amplicon sequencing
- take 16S reads -> map them to see who you have -> what do
  - what is function of this
  - do not use `expression` terms, use `representation`
  - finding more genes that could do it

#### Potential new research question:
- NASA dataset 1 - OSD #487
  - we can do a functional analysis (they didn't compare space vs earth - just looked at tissue types collectively)
  - they care about them getting diabetic foot ulcers
  - commonly get diabetes if you go to space
  - if diabetic foot ulcers that occur in space differ from those that naturally occuring? are theirs different? unique profile? would need a dataset that's amplicon sequencing for foot ulcers in control
  - NASA dataset has controls?
  - 1 sample sent to earth, 1 sample sent to space -> want to analyze how it progresses
  - challenges:
      - understanding very specific methodology
      - tough to compare to others
  - we'll still do:
      - taxonomic analysis
      - subset by things such as sex, age, etc
      - if have time, add functional component -> figure out how to clean data + what metadata is signficant or interested to us (ex. is space vs on earth actually significant)
      - might want to change continuous variables into categorical (ex. age 20-40 considered a `young` range) -> and compare diff groups
- NASA Dataset 2 - OSD #572
  - a longitudinal study ? looking at how microbiome progressed overtime - overtime in space + overtime when they returned
      - can predict it decreased overtime in space and then increased again when come back
  - looking at just skin microbiome analysis in space 
  - can only do amplicon sequencing (search for `16S` or `amplicon` in search bar from NASA website)
  - there are plant datasets too -> soil

NOTE: 1 more meeting until proposal deadline (proposal due on 23rd)

### Todos:
- [ ] find specific research Q w/ this dataset - make sure we really understand the experimental procedures + thinking about what we'd like to explore (3-4 aims) (ex. diversity analysis)
- [ ] read through proposal assignment on canvas
- [ ] be on lookout for team server
- [ ] online meeting next week
- [ ] archive all files on github (server isn't the most stable) - qza files

# Jan 31 2025 Team Meeting

### Agenda:
- discuss potential datasets

### Meeting Minutes:
- intros
- talked about intro papers
  - interested in depression paper, alcohol paper + ISS paper
  - can consider machine learning research Qs to create predictive models
- for next week's checkin
  - create github repository
  - come with list of research Qs
